### PR TITLE
Add: Network transfer graph to Linodes landing

### PR DESCRIPTION
**Pull request by Jskobos**
## Description

Add a progress bar representing network transfer usage to the top
of the Linodes landing view. When clicked, this opens a modal
with more details.

- Added props.children to the EntityHeader component since the existing
pattern wasn't working with this design.
- Basically re-implemented our Dialog component for the same reason.

## Note to Reviewers

Some things here are still a little rough, putting this up to get feedback before the demo. ~Look at the final commit to see the actual diff for this PR. Will merge/rebase once #7282 is merged.~

### CODE REVIEW

File: packages/manager/src/queries/accountTransfer.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import {
+  getNetworkUtilization,
+  NetworkUtilization
+} from '@linode/api-v4/lib/account';
+import { APIError } from '@linode/api-v4/lib/types';
+import { useQuery } from 'react-query';
+import { queryPresets } from './base';
+
+export const useAccountTransfer = () =>
+  useQuery<NetworkUtilization, APIError[]>(
```
Comment (johnwcallahan) [inline packages/manager/src/queries/accountTransfer.ts:10]: 💯

**Review by tiffwong — APPROVED** [review]

**Review by johnwcallahan — COMMENTED** [review]
I think this is looking good; only things I recommend fixing before the demo are:

- Usage > Quota: indication on Linodes Landing page
- Spacing of values > ~70% (see comment)

A few follow up items:

- Maybe use margin instead of padding, since the focus state makes the border run right against the docs button.
- Mobile could use some adjustment. 

Once this is approved we're probably going to want to use this component on the Object Storage and Account pages.

File: packages/manager/src/features/linodes/LinodesLanding/TransferDisplay.tsx [hunk 2]
```diff
@@ -14,13 +14,15 @@ const useStyles = makeStyles((theme: Theme) => ({
...
+    justifyContent: 'center',
     flexFlow: 'row nowrap'
   },
   wrapper: {
     border: 'none',
     backgroundColor: 'inherit',
-    cursor: 'pointer'
+    cursor: 'pointer',
+    marginRight: theme.spacing(4),
+    paddingRight: 0
```
Comment (tiffwong) [inline packages/manager/src/features/linodes/LinodesLanding/TransferDisplay.tsx:25]: This will fix the focus state running into the docs link

**Review by DevDW — APPROVED** [review]
Looks good -- in addition to the fixes for mobile, do we typically have a hover state for the "X" button for mini-modals like the one that pops up when you click "Monthly Network Transfer Pool"?

**Review by johnwcallahan — APPROVED** [review]
Looks great 👍

**Pull request merged**